cmd/sourced/compose/workdir/common.go

```diff
@@ -47,14 +51,29 @@ func (f *envFile) String() string {
 		gitbaseSiva = "true"
 	}
 
+	var gitbaseLimitCPU, gitcollectorLimitCPU string
+	dockerCPUs, err := dockerNumCPU()
+	if err != nil { // show warning
+		fmt.Println(err)
+	}
+	// apply gitbase resource limits only when docker runs without any global limits
+	// it's default behaviour on linux
+	if runtime.NumCPU() == dockerCPUs {
+		gitbaseLimitCPU = fmt.Sprintf("%.1f", float32(dockerCPUs)-0.1)
```

Comment:
Why not making the 0.1 configurable?
So if no limit passed there would be no limitations.

Reply:
as described in the PR this commit creates a foundation for limiting but doesn't change (much) how it works. `0.1` is only to try to avoid freezing the host system by default (I witnessed it on linux myself)
Any configuration or other rules should be implemented separately. We don't have any requirements for that yet.